### PR TITLE
MISUV-4026: Integrate new connector into existing service

### PR DESCRIPTION
**Pull request by ayubMercator**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: test/services/GetCalculationDetailsServiceSpec.scala [hunk 1]
```diff
@@ -111,19 +117,26 @@ class GetCalculationDetailsServiceSpec extends TestSuite {
   }
   ".getCalculationDetailsByCalcId" should {
 
-    "return a Right when successful" in {
+    "return a Right when successful and tax year >= 2024" in {
```
Comment (avelx) [inline test/services/GetCalculationDetailsServiceSpec.scala:120]: pls specify taxYear period instead, i.e. tax year 23-24

File: app/utils/TaxYear.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+object TaxYear {
+
+  def updatedFormat(year: String): String = {
```
Comment (avelx) [inline app/utils/TaxYear.scala:21]: we should not assume String will have require format + pls we need to have unit tests for this code

**Review by avelx — CHANGES_REQUESTED** [review]
minor changes requested

File: app/services/GetCalculationDetailsService.scala [hunk 3]
```diff
@@ -17,28 +17,35 @@
...
+        getCalculationDetailsByCalcId(nino, listOfCalculationDetails.head.calculationId, taxYear)
       case Left(desError) => Future.successful(Left(desError))
     }
   }
 
-  def getCalculationDetailsByCalcId(nino: String, calcId: String)(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
-    calculationDetailsConnector.getCalculationDetails(nino, calcId)
+  def getCalculationDetailsByCalcId(nino: String, calcId: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
+    taxYear match {
+      case Some(year) if year.toInt >= 2024 =>
```
[the remark this answers is not in the document]
Comment (avelx) [reply app/services/GetCalculationDetailsService.scala:45]: I would suggest to change => updatedFormat(year: String): Either[Error, String] and pattern match on updatedFormat result in line 44 instead

File: app/services/GetCalculationDetailsService.scala [hunk 4]
```diff
@@ -17,28 +17,51 @@
...
+  def getCalculationDetailsByCalcId(nino: String, calcId: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
+
+    convert(taxYear) match {
+      case Right(year) if year >= 2024 =>
+        calculationDetailsConnector.getCalculationDetails(updatedFormat(year.toString), nino, calcId)
+      case Right(_) => calculationDetailsConnectorLegacy.getCalculationDetails(nino, calcId)
+      case Left(error) => throw new RuntimeException(error)
+    }
+  }
+
```
Comment (avelx) [inline app/services/GetCalculationDetailsService.scala:51]: pls move these methods under ~/app/utils/*?

File: app/utils/TaxYear.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+class TaxYear {
```
Comment (avelx) [inline app/utils/TaxYear.scala:19]: pls remove empty class definition

File: app/services/GetCalculationDetailsService.scala [hunk 6]
```diff
@@ -17,28 +17,37 @@
...
       case Left(desError) => Future.successful(Left(desError))
     }
   }
 
-  def getCalculationDetailsByCalcId(nino: String, calcId: String)(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
-    calculationDetailsConnector.getCalculationDetails(nino, calcId)
+  def getCalculationDetailsByCalcId(nino: String, calcId: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[CalculationDetailResponse] = {
+
+    TaxYear.convert(taxYear) match {
+      case _ if taxYear.isEmpty => calculationDetailsConnectorLegacy.getCalculationDetails(nino, calcId)
```
Comment (avelx) [inline app/services/GetCalculationDetailsService.scala:46]: we might need to assume that this is and error in the future, as taxYear should be mandatory regardless what API we are calling; but we can leave this change for later

File: app/utils/TaxYear.scala (hunk 2 above)
Comment (ayubMercator) [reply app/utils/TaxYear.scala:31]: only called if tax year has been successfully parsed

**Review by avelx — APPROVED** [review]

**Pull request merged**